packages/core/src/api/apis/definitions/alert.ts

```diff
@@ -14,49 +14,47 @@
  * limitations under the License.
  */
 
+import { Color } from '@material-ui/lab';
 import ApiRef from '../ApiRef';
 
-/**
- * Mirrors the javascript Error class, for the purpose of
- * providing documentation and optional fields.
- */
-type Error = {
+type Alert = {
   name: string;
   message: string;
   stack?: string;
 };
 
 /**
- * Provides additional information about an error that was posted to the application.
+ * Provides additional information about an alert that was posted to the application.
  */
-export type ErrorContext = {
-  // If set to true, this error should not be displayed to the user. Defaults to false.
+export type AlertContext = {
+  // If set to true, this message should not be displayed to the user. Defaults to false.
   hidden?: boolean;
 };
 
 /**
- * The error API is used to report errors to the app, and display them to the user.
+ * The alert API is used to report alerts to the app, and display them to the user.
  *
- * Plugins can use this API as a method of displaying errors to the user, but also
- * to report errors for collection by error reporting services.
+ * Plugins can use this API as a method of displaying alerts to the user, but also
+ * to report alerts for collection by alert reporting services.
  *
- * If an error can be displayed inline, e.g. as feedback in a form, that should be
- * preferred over relying on this API to display the error. The main use of this api
- * for displaying errors should be for asynchronous errors, such as a failing background process.
+ * If an alert can be displayed inline, e.g. as feedback in a form, that should be
+ * preferred over relying on this API to display the alert. The main use of this api
+ * for displaying alerts should be for asynchronous alerts, such as a failing background process.
  *
- * Even if an error is displayed inline, it should still be reported through this api
+ * Even if an alert is displayed inline, it should still be reported through this api
  * if it would be useful to collect or log it for debugging purposes, but with
- * the hidden flag set. For example, an error arising from form field validation
+ * the hidden flag set. For example, an alert arising from form field validation
  * should probably not be reported, while a failed REST call would be useful to report.
  */
-export type ErrorApi = {
+
+export type AlertApi = {
   /**
-   * Post an error for handling by the application.
+   * Post an alert for handling by the application.
    */
-  post(error: Error, context?: ErrorContext);
+  post(alert: Alert, severity?: Color, context?: AlertContext);
```

Comment:
Should probably not use a MUI color - even a `lab` thing - in this long lived API and for this purpose :) While I appreciate that it's nifty, this could instead be an enumeration or dict in this file, e.g.

```
export type Severity = 'info' | 'warning' | 'error';
export type SeverityMapping: { [key in Severity]: { icon: SomeIconType, color: SomeColorType } };
const defaultSeverityMapping: SeverityMapping = {
 ...;
}
```

or something like that. Maybe the last one is actually a pretty good config object to pass to the instantiation of the alert API even. With a good and complete default of course, if none is given.

Reply:
yea was thinking of recreating out own but that was the typing on the severity prop on the snackbar. Figured it would be safer since we are directly assigning it to that prop.

Reply:
(the alert component does the icon and color mapping for you based on `severity`)

Reply:
agree though that `Color` is terribly named

Reply:
Whoa I did not realize. Well in that case, could it still be worth changing to not leak implementation details? Like what happens when that component is promoted out of lab? I'm not sure :)

Reply:
@Rugvip yea, i agree but it would be a bit hard for this use case because the first argument was modeled around the error object and there were examples where its being passed `new Error(...)`. In this cause you would have to append some outside `severity` property which is a bit awkward

Reply:
@wesleyyee don't care about existing usages in this repo at all when designing components and APIs imo. They're simple enough to update for now

Reply:
thats fair, tbh didn't give this much thought as i was just following the spec in the issue